### PR TITLE
Dlc on ruby 3.1 and shakapacker

### PR DESCRIPTION
**Pull request by barmintor**

### CODE REVIEW

File: config/deploy.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-lock '3.5.0'
+lock '3.14.1'
```
Comment (elohanlon) [inline config/deploy.rb:1]: Reminder here about the option to use `~>` (and maybe going ahead and committing the Capistrano version update you made locally too).

File: config/templates/database.template.yml [hunk 2]
```diff
@@ -1,10 +1,14 @@
-development:
+shared:
```
Comment (elohanlon) [inline config/templates/database.template.yml:1]: With the addition of this `shared` section, do you need `adapter`, `pool`, and `timeout` in the development and test sections?

**Review by elohanlon — APPROVED** [review]
Five stars! ⭐⭐⭐⭐⭐
Would review again!

Just had a couple of very minor comments, but glad to approve this PR!

File: config/templates/database.template.yml (hunk 2 above)
Comment (barmintor) [reply config/templates/database.template.yml:1]: The shared sections aren't working the way I expected them to, but I'd like to come back to it in a different commit.

File: config/deploy.rb (hunk 1 above)
Comment (barmintor) [reply config/deploy.rb:1]: Both good ideas - I'll add them to the other branch!

**Pull request merged**